src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -394,6 +444,104 @@ public EventBookingDTO requestBooking(final IsaacEventPageDTO event, final Regis
+     * requested.
+     *
+     * @param event - to reserve the user on
+     * @param users - to reserve on the event
+     * @return confirmation of reservation
+     */
+    public List<EventBookingDTO> requestReservations(final IsaacEventPageDTO event, final List<RegisteredUserDTO> users,
+                                                     final RegisteredUserDTO reservingUser)
+            throws EventDeadlineException, EmailMustBeVerifiedException, DuplicateBookingException,
+            SegueDatabaseException, EventIsFullException, EventGroupReservationLimitException {
+
+        // af599 TODO: Is it wise to do this before acquiring a database lock?
+        for (RegisteredUserDTO user : users) {
+            this.ensureValidBooking(event, user, true, BookingStatus.RESERVED);
+        }
+
+        List<EventBookingDTO> reservations = new ArrayList<>();
+        try {
+            // Obtain an exclusive database lock to lock the event
+            this.bookingPersistenceManager.acquireDistributedLock(event.getId());
+
+            // is there space on the event? Teachers don't count for student events.
+            // work out capacity information for the event at this moment in time.
+            // If there is no space, no reservations are made. Throw an exception and handle in EventsFacade.
+            this.ensureCapacity(event, users);
+
+            // Is the request for more reservations that this event allows?
+            this.enforceReservationLimit(event, users, reservingUser);
+
+            // IMPORTANT: Any non-ignorable exception past this point must roll back any reservation made thus far.
+            for (RegisteredUserDTO user : users) {
+                // attempt to book them on the event
+                EventBookingDTO reservation;
+
+                Calendar calendar = Calendar.getInstance();
+                calendar.add(Calendar.DAY_OF_MONTH, EVENT_RESERVATION_CLOSE_INTERVAL_DAYS);
+                Date reservationCloseDate = Stream.of(calendar.getTime(), event.getDate())
+                        .min(Comparator.comparing(Date::getTime))
+                        .orElseThrow(NoSuchElementException::new);
+
+                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+                Map<String, String> additionalEventInformation = new HashMap<>();
+                additionalEventInformation.put("reservationCloseDate", dateFormat.format(reservationCloseDate));
+
+                // attempt to book them on the event
+                if (this.hasBookingWithStatus(event.getId(), user.getId(), BookingStatus.CANCELLED)) {
+                    // if the user has previously cancelled we should let them book again.
+                    reservation = this.bookingPersistenceManager.updateBookingStatus(event.getId(), user.getId(),
+                            BookingStatus.RESERVED, additionalEventInformation);
```

Comment:
Updating a booking does not record the reservedBy user and so messes up the reservation limit counts a bit, and it could be confusing to have a reserved record with no reservedBy value.

Reply:
I think this goes back [to here](https://github.com/isaacphysics/isaac-api/pull/251#discussion_r400056774). I may have been rushed with that attribute but it did seem too much of an overkill to handle this differently.

Reply:
Would it not be just a matter of overloading updateBookingStatus and having the signature with less parameters call the one with more parameters?

Reply:
Indeed. Should be fixed in f80935d4.